fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/expressions/SelectExpression.java

```diff
@@ -173,11 +173,16 @@ private Set<CorrelationIdentifier> computeCorrelatedToWithoutChildren() {
     public SelectExpression translateCorrelations(@Nonnull final TranslationMap translationMap,
                                                   final boolean shouldSimplifyValues,
                                                   @Nonnull final List<? extends Quantifier> translatedQuantifiers) {
-        List<QueryPredicate> translatedPredicates =
-                predicates.stream()
+        final var definesOnlyIdentities = translationMap.definesOnlyIdentities();
+        final var translatedPredicates =
+                definesOnlyIdentities
+                ? predicates
+                : predicates.stream()
```

Comment:
Coverage information suggests that we're never going down this branch because `definesOnlyIdentities` is always `true`, at least for now. I'm not sure if we want to tackle this coverage gap in this PR. (Indeed, a lot of the coverage gaps identified by the coverage check are, I think, caused by a lack of calls to `translateCorrelations`, especially for plans.)

Reply:
I agree that this could do with better unit testing, but not in this PR.

Reply:
@hatyo next PR will utilize this code path with a non-identity translation map.